### PR TITLE
Update GitLab CI configuration

### PR DESCRIPTION
**Pull request by tmaczukin**
Improves integration with GitLab CI.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,25 @@
+export CGO_ENABLED ?= 0
+
+VERSION ?= $(shell git describe --tags 2>/dev/null | sed -e 's/^v//g' || echo "dev")
+REVISION ?= $(shell git rev-parse --short HEAD || echo "unknown")
+DATE ?= $(shell date +%Y-%m-%dT%H:%M:%S%:z)
+LDFLAGS = -s -w \
+          -X github.com/gomeeseeks/meeseeks-box/version.Version=$(VERSION) \
+          -X github.com/gomeeseeks/meeseeks-box/version.Commit=$(REVISION) \
+          -X github.com/gomeeseeks/meeseeks-box/version.Date=$(DATE)
```
Comment (pcarranza) [inline Makefile]: This, and the docker image creation (darwin, amd64-linux and armv6) are handled with goreleaser. I would like to keep it this way. This is why the makefile is so simple.

I would suggest you to try to run `make snapshot` after getting [goreleaser](https://goreleaser.com/) installed and see how we can leverage this tool as the building and releasing one.

One of the core benefits of using goreleaser is that there is an open issue to [make it provide multi arch docker images](https://github.com/goreleaser/goreleaser/issues/530), which will simplify distribution quite a lot.

File: Makefile [hunk 2]
```diff
@@ -15,3 +28,13 @@ release: test
...
+
+CI_REGISTRY_IMAGE ?= gomeeseeks/meeseeks-box
+
+build_docker:
+	docker build -t $(CI_REGISTRY_IMAGE):$(VERSION) -f Dockerfile .
+	if [ -n "$$CI_REGISTRY" ]; then \
+	  docker login --username $$CI_REGISTRY_USER --password $$CI_REGISTRY_PASSWORD $$CI_REGISTRY && \
+	  docker push $(CI_REGISTRY_IMAGE):$(VERSION) && \
+	  docker logout $$CI_REGISTRY; \
+	fi
```
Comment (pcarranza) [inline Makefile]: Check my previous comment, this should not be necessary as part of the makefile (the docker build at least), then we could extract the gitlab-ci specifics into the gitlab-ci file leaving this one provider agnostic.

**Comment by tmaczukin** [comment]
@pcarranza Got it! I'll transform this to make usage of goreleaser :)

**Comment by pcarranza** [comment]
Closing for now, please reopen with an up to date gitlab-ci file.

**Pull request closed without merging**